Sample Application/SampleApplication/Models/Authenticator.swift

```diff
@@ -0,0 +1,75 @@
+// 
+//  Authenticator.swift
+//  SampleApplication
+//
+//  Created by Okhan Okbay on 23/05/2023.
+//
+
+import CheckoutNetwork
+import Foundation
+
+#if canImport(CheckoutCardManagement)
+import CheckoutCardManagement
+#elseif canImport(CheckoutCardManagementStub)
+import CheckoutCardManagementStub
+#endif
+
+/*
+ Where documented, you are required to perform StrongCustomAuthentication (SCA)
+ in accordance to your company's policies and processes.
+ The sample project integrates directly with CKO backend for token creation in order to enable the flows,
+ but in a live application this is expected to occur from your backend to CKO backend and not be performed via your application.
+ Hence you can assess CheckoutNetwork as a network wrapper that you will not use.
+ */
+
+struct Authenticator {
+    private let networkClient: CheckoutNetworkClient = .init()
+    
+    func authenticateUser(checkDeviceOwnership: Bool = true,
+                          completionHandler: @escaping ((Result<String, CardManagementError>) -> Void)) {
+        
+        guard checkDeviceOwnership else {
+            authenticate(completionHandler: completionHandler)
+            return
+        }
+        
+        AuthenticationValidator.isDeviceOwner { isDeviceOwner in
+            guard isDeviceOwner else {
```

Comment:
I would again rename these methods and properties with references to SCA. A version may be:
```Swift
AuthenticationValidator.performSCA { scaValid in
```

Reply:
Checking device ownership _may_ be a type of SCA. We are not pitching SCA solutions and do not have expertise currently in what valid SCA processes mean. We do know that it may depend on company processes & setup, as well as risk aversion.

Writing code that says "check device ownership" shows that you need to check device ownership.

Writing code that says "perform SCA" shows that you need to perform SCA. Now that **our sample interpretation** of SCA is to perform a device check, this is a different aspect. We also document at the source of asking for device ownership that it is not representative of the clients expected SCA flow.

Reply:
good point! still doesn't prevent the confusion over the differentiation points of `SCA` and `device ownership` for someone who only follows the code but improves it a little bit. I kept the function name in `AuthenticationValidator` because `SCA` means checking the device ownership for us in this case and the code tells that story